### PR TITLE
(CDAP-1712) Added support for configurable column family

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/table/Table.java [hunk 1]
```diff
@@ -42,6 +42,12 @@
   String PROPERTY_READLESS_INCREMENT = "dataset.table.readless.increment";
 
   /**
+   * Property set to configure name of the column family. The property only applies to implementations that support
+   * it. If not set, a default column family will be used.
+   */
+  String COLUMN_FAMILY = "dataset.table.column.family";
```
Comment (albertshau) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/table/Table.java]: should we name it PROPERTY_COLUMN_FAMILY to be consistent with other properties here?

File: cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java [hunk 2]
```diff
@@ -98,8 +105,11 @@ public static void afterClass() throws Exception {
   @Override
   protected BufferingTable getTable(String name, ConflictDetection conflictLevel) throws Exception {
     // ttl=-1 means "keep data forever"
-    return
-      new HBaseTable(name, ConflictDetection.valueOf(conflictLevel.name()), testHBase.getConfiguration(), true);
+    DatasetSpecification spec = DatasetSpecification.builder(name, "foo")
+      .property(Table.PROPERTY_READLESS_INCREMENT, "true")
+      .property("conflict.level", conflictLevel.name())
```
Comment (albertshau) [inline cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java]: i know it wasn't introduced here, but should we also make this a constant in Table while we're at it?

**Comment by albertshau** [comment]
couple minor questions, otherwise lgtm

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/table/Table.java (hunk 1 above)
Comment (anew) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/table/Table.java]: +1

File: cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java (hunk 2 above)
Comment (anew) [inline cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java]: +1

**Comment by anew** [comment]
Agreed with Albert's comments. Other than that LGTM

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/table/Table.java (hunk 1 above)
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/table/Table.java]: sure. Missed that.

File: cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java (hunk 2 above)
Comment (chtyim) [inline cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableTest.java]: Sure. Was trying to keep this PR small.

**Comment by chtyim** [comment]
Resolved conflicts with develop and addressed comments.

**Comment by chtyim** [comment]
Build running in http://builds.cask.co/browse/CDAP-DUT1011-2

**Comment by chtyim** [comment]
Build passed. Merging.

**Pull request merged**